qp/pdf.py

```diff
@@ -76,38 +108,35 @@ def interpolate(self, number=100):
         interpolated : ndarray, float
             The interpolated points.
 
+        Comments
+        --------
+        Extrapolation is linear until values are outside [0., 1.]; outside this range, extrapolation returns 0.
+
         Notes
         -----
-
+        Example:
+        > x, y = p.approximate(np.linspace(-1., 1., 100))
         """
-        if self.interpolator is None:
-
-            if self.quantiles is None:
-                self.quantiles = self.quantize()
-            self.difs = self.quantiles[1:]-self.quantiles[:-1]
-            self.mids = (self.quantiles[1:]+self.quantiles[:-1])/2.
-            self.quantvals = (1.0/(len(self.quantiles)+1))/self.difs
-
-            print("Creating interpolator")
-            self.interpolator = spi.interp1d(self.mids, self.quantvals)
 
-        grid = np.linspace(min(self.mids), max(self.mids), number)
-        print("Grid: ", grid)
-        interpolated = self.interpolator(grid)
-        print("Result: ", interpolated)
+        if self.interpolator is None:
+            self.interpolator = self.interpolate()
+        #print("Grid: ", x)
+        interpolated = self.interpolator(points)
+        interpolated[interpolated<0.] = 0.
```

Comment:
I think this is good. Can you think of any situation where you would use the `self.interpolator` directly rather than using teh `approximate` method? I can't... In which case, I'd suggest not returning the `interpolator` from the `interpolate` method, to avoid confusion. WHat do you think?